### PR TITLE
test: add logger edge case tests

### PR DESCRIPTION
**Pull request by mthalman**
## Summary

Add comprehensive edge case tests for the logger's serializeArg function and logger invocations with unusual argument types.

## Test Coverage Added

- **Circular references**: deeply nested, circular arrays, mutually circular objects
- **Large objects**: deeply nested (100 levels), objects with 1000 properties, large arrays
- **Primitive edge cases**: undefined, null, NaN, Infinity, -Infinity, 0/-0, booleans
- **Error objects**: TypeError, RangeError, custom properties, empty message
- **Functions**: named functions, arrow functions (String() fallback)
- **Symbols**: named symbols, well-known symbols (Symbol.iterator)
- **Empty values**: empty string, empty array, empty object
- **Date objects**: valid dates (ISO string), invalid dates
- **toJSON**: objects with custom toJSON methods
- **RegExp, Map, Set**: JSON.stringify behavior verification
- **BigInt**: String() fallback for BigInt values
- **Mixed nested types**: undefined values in objects, null holes in arrays, functions in objects
- **Logger integration**: multiple edge-case args, symbol/BigInt/Error/function args, uninitialized channel

### CODE REVIEW

File: packages/core/src/test/logger.test.ts [hunk 1]
```diff
@@ -136,5 +137,275 @@ describe('logger', () => {
...
+    });
+
+    describe('mixed nested types', () => {
+      it('should handle objects with undefined values (omitted by JSON)', () => {
+        const obj = { a: 1, b: undefined, c: 'hello' };
+        const result = serializeArg(obj);
+        expect(result).toBe('{"a":1,"c":"hello"}');
+      });
+
+      it('should handle arrays with null holes', () => {
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/core/src/test/logger.test.ts]: Fixed — renamed to 'should handle arrays with explicit null values' per your suggestion.

File: packages/core/src/test/logger.test.ts [hunk 2]
```diff
@@ -136,5 +137,275 @@ describe('logger', () => {
...
+        const obj: any = { a: { b: { c: {} } } };
+        obj.a.b.c.root = obj;
+        expect(() => serializeArg(obj)).not.toThrow();
+        expect(serializeArg(obj)).toBe('[object Object]');
+      });
+
+      it('should handle circular arrays', () => {
+        const arr: any[] = [1, 2];
+        arr.push(arr);
+        expect(() => serializeArg(arr)).not.toThrow();
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/core/src/test/logger.test.ts:154]: This test passes reliably. V8/Node's Array.prototype.join handles circular references by substituting an empty string for the cyclic element rather than recursing infinitely. Verified by running the test in isolation: 1 passed, 0 failed. No change needed.
